// @uiw/react-checkbox/cjs/Checkbox.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _reactRadio = require("@uiw/react-radio");

var _Group = require("./Group");

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["className", "prefixCls", "type", "indeterminate", "disabled", "value"];

function InternalCheckbox(props, ref) {
  var className = props.className,
      _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-checkbox' : _props$prefixCls,
      _props$type = props.type,
      type = _props$type === void 0 ? 'checkbox' : _props$type,
      _props$indeterminate = props.indeterminate,
      indeterminate = _props$indeterminate === void 0 ? false : _props$indeterminate,
      _props$disabled = props.disabled,
      disabled = _props$disabled === void 0 ? false : _props$disabled,
      _props$value = props.value,
      value = _props$value === void 0 ? '' : _props$value,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var cls = [className, indeterminate && 'indeterminate'].filter(Boolean).join(' ').trim();
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactRadio.RadioAbstract, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    ref: ref
  }, other), {}, {
    type: type,
    prefixCls: prefixCls,
    disabled: disabled,
    value: value,
    className: cls
  }));
}

var Checkbox = /*#__PURE__*/_react["default"].forwardRef(InternalCheckbox);

Checkbox.Group = _Group.CheckboxGroup;
var _default = Checkbox;
exports["default"] = _default;
module.exports = exports.default;
