// @uiw/react-collapse/cjs/Collapse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _Panel = _interopRequireDefault(require("./Panel"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "children", "accordion", "bordered", "showArrow", "activeKey", "onChange"];

function toArray(activeKey) {
  var currentActiveKey = activeKey;

  if (!Array.isArray(currentActiveKey)) {
    currentActiveKey = currentActiveKey ? [currentActiveKey] : [];
  }

  return currentActiveKey;
}

function InternalCollapse(props, ref) {
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-collapse' : _props$prefixCls,
      className = props.className,
      children = props.children,
      _props$accordion = props.accordion,
      accordion = _props$accordion === void 0 ? false : _props$accordion,
      bordered = props.bordered,
      _props$showArrow = props.showArrow,
      showArrow = _props$showArrow === void 0 ? true : _props$showArrow,
      propsActiveKey = props.activeKey,
      onChange = props.onChange,
      resetProps = (0, _objectWithoutProperties2["default"])(props, _excluded);

  var _useState = (0, _react.useState)(toArray(propsActiveKey)),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      activeKey = _useState2[0],
      setActiveKey = _useState2[1];

  var cls = [prefixCls, className, bordered ? 'w-noborder' : null].filter(Boolean).join(' ').trim();

  function onItemClick(key) {
    var keys = activeKey;

    if (accordion) {
      keys = keys[0] === key ? [] : [key];
    } else {
      keys = (0, _toConsumableArray2["default"])(keys);
      var index = keys.indexOf(key);
      var isActive = index > -1;

      if (isActive) {
        keys.splice(index, 1);
      } else {
        keys.push(key);
      }
    }

    setActiveKey(keys);
  }

  (0, _react.useMemo)(function () {
    if (propsActiveKey !== activeKey) {
      setActiveKey(toArray(propsActiveKey));
    }
  }, [propsActiveKey]);
  (0, _react.useMemo)(function () {
    if (propsActiveKey !== activeKey) {
      onChange && onChange(activeKey);
    }
  }, [activeKey, propsActiveKey]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    className: cls
  }, resetProps), {}, {
    ref: ref,
    children: _react["default"].Children.map(children, function (child, index) {
      // 如果没有密钥提供，请使用面板顺序作为默认密钥
      var key = child.key || String(index);
      var disabled = child.props.disabled;
      var isActive = false;

      if (accordion) {
        // 手风琴模式下默认选择第一个
        isActive = activeKey[0] === key;
      } else {
        isActive = activeKey.indexOf(key) > -1;
      }

      var childProps = (0, _objectSpread2["default"])({
        prefixCls: prefixCls,
        isActive: isActive,
        disabled: disabled,
        showArrow: showArrow,
        onItemClick: disabled ? function () {} : function () {
          return onItemClick(key);
        }
      }, child.props);
      return /*#__PURE__*/_react["default"].cloneElement(child, childProps);
    })
  }));
}

var Collapse = /*#__PURE__*/_react["default"].forwardRef(InternalCollapse);

Collapse.Panel = _Panel["default"];
var _default = Collapse;
exports["default"] = _default;
module.exports = exports.default;
