// @uiw/react-collapse/cjs/Panel.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Panel;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _reactTransitionGroup = require("react-transition-group");

var _reactIcon = _interopRequireDefault(require("@uiw/react-icon"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "icon", "children", "isActive", "onItemClick", "disabled", "showArrow", "header", "extra"];

function Panel() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-collapse' : _props$prefixCls,
      className = props.className,
      _props$icon = props.icon,
      icon = _props$icon === void 0 ? 'down' : _props$icon,
      _children = props.children,
      isActive = props.isActive,
      onItemClick = props.onItemClick,
      _props$disabled = props.disabled,
      disabled = _props$disabled === void 0 ? false : _props$disabled,
      showArrow = props.showArrow,
      header = props.header,
      extra = props.extra,
      resetProps = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var cls = [prefixCls ? "".concat(prefixCls, "-item") : null, className, isActive ? "".concat(prefixCls, "-active") : null, disabled ? "".concat(prefixCls, "-disabled") : null].filter(Boolean).join(' ').trim();
  var iconRender = typeof icon === 'string' ? /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactIcon["default"], {
    type: icon
  }) : icon;

  var childStyle = function childStyle(child) {
    return Object.assign({}, child && child.props ? child.props.style : {}, {
      transitionDuration: '300ms'
    });
  };

  function getInstance(status, instance) {
    if (!instance) {
      return;
    }

    if (status === 'exited' || status === 'exiting') {
      instance.style.height = '1px';
    }

    if (status === 'entered' || status === 'entering') {
      instance.style.height = "".concat(instance.scrollHeight, "px");
    }
  }

  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    className: cls
  }, resetProps), {}, {
    children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
      className: "".concat(prefixCls, "-header"),
      onClick: onItemClick,
      children: [showArrow && iconRender, /*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
        className: "".concat(prefixCls, "-title"),
        children: header
      }), extra && /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        className: "".concat(prefixCls, "-extra"),
        children: extra
      })]
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactTransitionGroup.CSSTransition, {
      "in": isActive,
      unmountOnExit: false,
      timeout: 300,
      classNames: "".concat(prefixCls, "-panel"),
      children: function children(status) {
        return /*#__PURE__*/_react["default"].cloneElement( /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
          children: _children
        }), {
          className: "".concat(prefixCls, "-panel"),
          style: childStyle(_children),
          ref: function ref(e) {
            return getInstance(status, e);
          }
        });
      }
    })]
  }));
}

module.exports = exports.default;
