// @uiw/react-descriptions/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {};
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _DescriptionsItem = _interopRequireWildcard(require("./DescriptionsItem"));

Object.keys(_DescriptionsItem).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _DescriptionsItem[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _DescriptionsItem[key];
    }
  });
});

var _Row = _interopRequireDefault(require("./Row"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "title", "bordered", "column", "size", "colon", "children", "layout"];

var generateChildrenRows = function generateChildrenRows(children, column) {
  var rows = [];
  var columns = null;
  var leftSpans;
  children.forEach(function (node, index) {
    var itemNode = node;

    if (!columns) {
      leftSpans = column;
      columns = [];
      rows.push(columns);
    } // Always set last span to align the end of Descriptions


    var lastItem = index === children.length - 1;

    if (lastItem) {
      itemNode = /*#__PURE__*/_react["default"].cloneElement(itemNode, {
        span: leftSpans
      });
    } // Calculate left fill span


    var _itemNode$props$span = itemNode.props.span,
        span = _itemNode$props$span === void 0 ? 1 : _itemNode$props$span;
    columns.push(itemNode);
    leftSpans -= span;

    if (leftSpans <= 0) {
      columns = null;
    }
  });
  return rows;
};

function InternalDescriptions(props, ref) {
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-descriptions' : _props$prefixCls,
      className = props.className,
      title = props.title,
      bordered = props.bordered,
      _props$column = props.column,
      column = _props$column === void 0 ? 3 : _props$column,
      size = props.size,
      _props$colon = props.colon,
      colon = _props$colon === void 0 ? true : _props$colon,
      children = props.children,
      _props$layout = props.layout,
      layout = _props$layout === void 0 ? 'horizontal' : _props$layout,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var cls = [prefixCls, className, prefixCls && layout ? "".concat(prefixCls, "-").concat(layout) : null, bordered ? "".concat(prefixCls, "-bordered") : null, size ? "".concat(prefixCls, "-").concat(size) : null].filter(Boolean).join(' ').trim();

  var cloneChildren = _react["default"].Children.toArray(children);

  var childs = generateChildrenRows(cloneChildren, column);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
    className: cls,
    ref: ref,
    children: /*#__PURE__*/(0, _jsxRuntime.jsxs)("table", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, other), {}, {
      children: [title && /*#__PURE__*/(0, _jsxRuntime.jsx)("caption", {
        className: "".concat(prefixCls, "-title"),
        children: title
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)("tbody", {
        children: childs.map(function (child, index) {
          return /*#__PURE__*/(0, _jsxRuntime.jsx)(_Row["default"], {
            prefixCls: prefixCls,
            bordered: bordered,
            colon: colon,
            column: column,
            layout: layout,
            children: child
          }, index);
        })
      })]
    }))
  });
}

var Descriptions = /*#__PURE__*/_react["default"].forwardRef(InternalDescriptions);

Descriptions.Item = _DescriptionsItem["default"];
var _default = Descriptions;
exports["default"] = _default;
