// @uiw/react-descriptions/cjs/Row.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Row;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _react = _interopRequireWildcard(require("react"));

var _Cell = _interopRequireDefault(require("./Cell"));

var _jsxRuntime = require("react/jsx-runtime");

function Row() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var prefixCls = props.prefixCls,
      layout = props.layout,
      bordered = props.bordered,
      column = props.column,
      colon = props.colon,
      _props$children = props.children,
      children = _props$children === void 0 ? [] : _props$children;

  function handleCell(isHead) {
    return children.map(function (child, index) {
      return /*#__PURE__*/(0, _react.createElement)(_Cell["default"], (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, child.props), {}, {
        prefixCls: prefixCls,
        key: index,
        column: column,
        tagName: isHead ? 'th' : 'td',
        isLastCell: children.length - 1 === index,
        layout: layout,
        colon: colon,
        bordered: bordered
      }), isHead ? child.props.label : child.props.children);
    });
  }

  var cls = prefixCls ? "".concat(prefixCls, "-row") : '';
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_react.Fragment, {
    children: [layout === 'vertical' && /*#__PURE__*/(0, _jsxRuntime.jsx)("tr", {
      className: cls,
      children: handleCell(true)
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)("tr", {
      className: cls,
      children: handleCell()
    })]
  });
}

module.exports = exports.default;
