// @uiw/react-progress/cjs/Line.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _utils = require("./utils");

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "style", "className", "showText", "percent", "format", "strokeWidth", "width", "status"];

var Line = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Line, _React$Component);

  var _super = (0, _createSuper2["default"])(Line);

  function Line() {
    (0, _classCallCheck2["default"])(this, Line);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(Line, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          prefixCls = _this$props.prefixCls,
          style = _this$props.style,
          className = _this$props.className,
          showText = _this$props.showText,
          percent = _this$props.percent,
          format = _this$props.format,
          strokeWidth = _this$props.strokeWidth,
          width = _this$props.width,
          status = _this$props.status,
          resetProps = (0, _objectWithoutProperties2["default"])(_this$props, _excluded);
      var cls = [prefixCls, className, "".concat(prefixCls, "-line"), showText ? "".concat(prefixCls, "-show-text") : null, status ? "".concat(prefixCls, "-status-").concat(status) : null, parseInt(percent.toString(), 10) >= 100 ? "".concat(prefixCls, "-status-success") : null].filter(Boolean).join(' ').trim();
      var progressInfo;

      if (showText) {
        var progressStatus = parseInt(percent.toString(), 10) >= 100 && !('status' in this.props) ? 'success' : status;
        var percentView = "".concat(percent, "%");

        if (progressStatus === 'exception') {
          percentView = /*#__PURE__*/(0, _jsxRuntime.jsx)(_utils.IconProgress, {
            type: "circle-close"
          });
        } else if (progressStatus === 'success') {
          percentView = /*#__PURE__*/(0, _jsxRuntime.jsx)(_utils.IconProgress, {
            type: "circle-check"
          });
        }

        progressInfo = /*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
          className: "".concat(prefixCls, "-text"),
          children: format ? format(percent) : percentView
        });
      }

      var percentStyle = {
        width: "".concat(percent, "%"),
        height: strokeWidth
      };
      return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
        className: cls,
        style: style
      }, resetProps), {}, {
        children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
          className: "".concat(prefixCls, "-bar"),
          children: /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
            className: "".concat(prefixCls, "-inner"),
            children: /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
              className: "".concat(prefixCls, "-bg"),
              style: percentStyle
            })
          })
        }), progressInfo]
      }));
    }
  }]);
  return Line;
}(_react["default"].Component);

exports["default"] = Line;
(0, _defineProperty2["default"])(Line, "defaultProps", {
  prefixCls: 'w-progress',
  showText: true,
  // 是否显示进度条文字内容
  percent: 0,
  // 百分比（必填）
  width: 126,
  // 圆圈进度条画布宽度
  strokeWidth: 6 // 进度条大小设置

});
module.exports = exports.default;
