// @uiw/react-form/cjs/FormItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _reactGrid = require("@uiw/react-grid");

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "style", "label", "labelFor", "labelClassName", "labelStyle", "help", "inline", "initialValue", "validator", "hasError"];

var FormItem = /*#__PURE__*/function (_React$PureComponent) {
  (0, _inherits2["default"])(FormItem, _React$PureComponent);

  var _super = (0, _createSuper2["default"])(FormItem);

  function FormItem() {
    (0, _classCallCheck2["default"])(this, FormItem);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(FormItem, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          prefixCls = _this$props.prefixCls,
          className = _this$props.className,
          style = _this$props.style,
          label = _this$props.label,
          labelFor = _this$props.labelFor,
          labelClassName = _this$props.labelClassName,
          labelStyle = _this$props.labelStyle,
          help = _this$props.help,
          inline = _this$props.inline,
          initialValue = _this$props.initialValue,
          validator = _this$props.validator,
          hasError = _this$props.hasError,
          otherProps = (0, _objectWithoutProperties2["default"])(_this$props, _excluded);
      var cls = [prefixCls, className, hasError ? "".concat(prefixCls, "-error") : null].filter(Boolean).join(' ').trim();
      var labelCls = ['w-form-label', labelClassName].filter(Boolean).join(' ').trim();

      if (inline) {
        return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
          className: cls,
          style: style
        }, otherProps), {}, {
          children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)(_reactGrid.Row, {
            children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_reactGrid.Col, {
              fixed: true,
              className: labelCls,
              children: /*#__PURE__*/(0, _jsxRuntime.jsx)("label", {
                style: labelStyle,
                htmlFor: labelFor,
                children: label
              })
            }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactGrid.Col, {
              className: "w-form-row",
              children: this.props.children
            })]
          }), help && /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactGrid.Row, {
            children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactGrid.Col, {
              className: "w-form-help",
              children: help
            })
          })]
        }));
      }

      return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
        className: cls,
        style: style
      }, otherProps), {}, {
        children: [label && /*#__PURE__*/(0, _jsxRuntime.jsx)("label", {
          className: labelCls,
          style: labelStyle,
          htmlFor: labelFor,
          children: label
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactGrid.Col, {
          className: "w-form-row",
          children: this.props.children
        }), help && /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
          className: "w-form-help",
          children: help
        })]
      }));
    }
  }]);
  return FormItem;
}(_react["default"].PureComponent);

exports["default"] = FormItem;
(0, _defineProperty2["default"])(FormItem, "defaultProps", {
  prefixCls: 'w-form-item'
});
module.exports = exports.default;
