// @uiw/react-radio/cjs/RadioGroup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "name", "value", "onChange", "children"];

var _default = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-radio-group' : _props$prefixCls,
      className = props.className,
      name = props.name,
      value = props.value,
      onChange = props.onChange,
      children = props.children,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, other), {}, {
    ref: ref,
    className: [prefixCls, className].filter(Boolean).join(' ').trim(),
    children: _react["default"].Children.toArray(children).map(function (child) {
      if (!child) return;
      if (! /*#__PURE__*/_react["default"].isValidElement(child)) return child;
      return /*#__PURE__*/_react["default"].cloneElement(child, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, child.props || {}), {
        checked: child.props.value === value,
        name: name,
        onChange: onChange
      }));
    })
  }));
});

exports["default"] = _default;
module.exports = exports.default;
