// @uiw/react-radio/cjs/RadioAbstract.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RadioAbstract = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "type", "disabled", "value", "className", "style", "children", "size", "checked", "onChange"];

var RadioAbstract = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-radio' : _props$prefixCls,
      _props$type = props.type,
      type = _props$type === void 0 ? 'radio' : _props$type,
      _props$disabled = props.disabled,
      disabled = _props$disabled === void 0 ? false : _props$disabled,
      _props$value = props.value,
      value = _props$value === void 0 ? '' : _props$value,
      className = props.className,
      style = props.style,
      children = props.children,
      size = props.size,
      _props$checked = props.checked,
      prChecked = _props$checked === void 0 ? false : _props$checked,
      onChange = props.onChange,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);

  var _useState = (0, _react.useState)(prChecked),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      checked = _useState2[0],
      setChecked = _useState2[1];

  var _useState3 = (0, _react.useState)(),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      prevChecked = _useState4[0],
      setPrevChecked = _useState4[1];

  if (prChecked !== prevChecked) {
    setPrevChecked(prChecked);
  }

  (0, _react.useMemo)(function () {
    if (prChecked !== prevChecked) {
      setChecked(prChecked);
    }
  }, [prevChecked]);
  var cls = [prefixCls, className, disabled ? 'disabled' : null, size ? "".concat(prefixCls, "-").concat(size) : null].filter(Boolean).join(' ').trim();
  (0, _react.useMemo)(function () {
    if (checked !== props.checked) {
      setChecked(!!props.checked);
    }
  }, [props.checked]);

  function handleChange(e) {
    e.persist();
    setChecked(e.target.checked);
    onChange && onChange(e);
  }

  var label = children || value;
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("label", {
    className: cls,
    style: style,
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("input", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, other), {}, {
      type: type,
      disabled: disabled,
      value: value
    })), {}, {
      checked: checked,
      onChange: handleChange,
      ref: ref
    })), label && /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: "".concat(prefixCls, "-text"),
      children: label
    })]
  });
});

exports.RadioAbstract = RadioAbstract;
