// @uiw/react-back-top/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _utils = require("./utils");

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "content", "children", "offsetTop", "fixed", "speed", "showBelow", "clickable"];

var _default = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-back-top' : _props$prefixCls,
      className = props.className,
      content = props.content,
      children = props.children,
      _props$offsetTop = props.offsetTop,
      offsetTop = _props$offsetTop === void 0 ? 0 : _props$offsetTop,
      _props$fixed = props.fixed,
      fixed = _props$fixed === void 0 ? true : _props$fixed,
      _props$speed = props.speed,
      speed = _props$speed === void 0 ? 100 : _props$speed,
      _props$showBelow = props.showBelow,
      showBelow = _props$showBelow === void 0 ? 1 : _props$showBelow,
      _props$clickable = props.clickable,
      clickable = _props$clickable === void 0 ? true : _props$clickable,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var topShowBelow = !fixed ? 0 : showBelow || 0;

  var _useState = (0, _react.useState)(0),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      percent = _useState2[0],
      setPercent = _useState2[1];

  var _useState3 = (0, _react.useState)(0),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      current = _useState4[0],
      setCurrent = _useState4[1];

  var visible = percent >= topShowBelow;
  var cls = [prefixCls, className, !fixed ? 'no-fixed' : null, visible ? "".concat(prefixCls, "-show") : null, !visible ? "".concat(prefixCls, "-hide") : null].filter(Boolean).join(' ').trim();
  (0, _react.useEffect)(function () {
    window && window.addEventListener('scroll', onScroll);
    return function () {
      window && window.removeEventListener('scroll', onScroll);
    };
  });

  function onScroll() {
    setPercent((0, _utils.getScrollPercent)(offsetTop));
    setCurrent((0, _utils.getScrollTop)());
  }

  function scrollToTop() {
    if (typeof offsetTop === 'number' && typeof speed === 'number' && typeof current === 'number') {
      (0, _utils.scrollToAnimate)(offsetTop, speed, current);
    }
  }

  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    onClick: function onClick() {
      return clickable && scrollToTop();
    },
    className: cls
  }, other), {}, {
    ref: ref,
    children: [content, typeof children !== 'function' ? children : children({
      percent: percent,
      current: current,
      scrollToTop: scrollToTop
    })]
  }));
});

exports["default"] = _default;
module.exports = exports.default;
