// @uiw/react-grid/cjs/Row.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Row = Row;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "gutter", "justify", "align"];

function Row() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-row' : _props$prefixCls,
      className = props.className,
      _props$gutter = props.gutter,
      gutter = _props$gutter === void 0 ? 0 : _props$gutter,
      justify = props.justify,
      align = props.align,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var cls = [prefixCls, className, align ? "".concat(prefixCls, "-align-").concat(align) : null, justify ? "".concat(prefixCls, "-justify-").concat(justify) : null].filter(Boolean).join(' ').trim();
  var gutterStyl = !gutter ? {} : {
    paddingLeft: gutter / 2,
    paddingRight: gutter / 2
  };
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, other), {}, {
    className: cls,
    children: _react["default"].Children.toArray(props.children).map(function (child) {
      if (! /*#__PURE__*/_react["default"].isValidElement(child)) return child;
      return /*#__PURE__*/_react["default"].cloneElement(child, Object.assign({}, child.props, {
        style: (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, child.props.style), gutterStyl)
      }));
    })
  }));
}
