// @uiw/react-modal/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _reactOverlay = _interopRequireDefault(require("@uiw/react-overlay"));

var _reactButton = _interopRequireDefault(require("@uiw/react-button"));

var _reactIcon = _interopRequireDefault(require("@uiw/react-icon"));

var _utils = require("@uiw/utils");

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "children", "useButton", "usePortal", "autoFocus", "isOpen", "title", "cancelText", "cancelButtonProps", "confirmButtonProps", "content", "confirmText", "type", "icon", "maxWidth", "minWidth", "width", "isCloseButtonShown", "onCancel", "onConfirm", "bodyStyle"];

var _default = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-modal' : _props$prefixCls,
      className = props.className,
      children = props.children,
      _props$useButton = props.useButton,
      useButton = _props$useButton === void 0 ? true : _props$useButton,
      _props$usePortal = props.usePortal,
      usePortal = _props$usePortal === void 0 ? true : _props$usePortal,
      _props$autoFocus = props.autoFocus,
      autoFocus = _props$autoFocus === void 0 ? false : _props$autoFocus,
      _props$isOpen = props.isOpen,
      _ = _props$isOpen === void 0 ? false : _props$isOpen,
      title = props.title,
      cancelText = props.cancelText,
      cancelButtonProps = props.cancelButtonProps,
      confirmButtonProps = props.confirmButtonProps,
      content = props.content,
      _props$confirmText = props.confirmText,
      confirmText = _props$confirmText === void 0 ? 'чбошод' : _props$confirmText,
      _props$type = props.type,
      type = _props$type === void 0 ? 'light' : _props$type,
      icon = props.icon,
      _props$maxWidth = props.maxWidth,
      maxWidth = _props$maxWidth === void 0 ? 500 : _props$maxWidth,
      _props$minWidth = props.minWidth,
      minWidth = _props$minWidth === void 0 ? 320 : _props$minWidth,
      width = props.width,
      _props$isCloseButtonS = props.isCloseButtonShown,
      isCloseButtonShown = _props$isCloseButtonS === void 0 ? true : _props$isCloseButtonS,
      _props$onCancel = props.onCancel,
      onCancel = _props$onCancel === void 0 ? _utils.noop : _props$onCancel,
      _props$onConfirm = props.onConfirm,
      onConfirm = _props$onConfirm === void 0 ? _utils.noop : _props$onConfirm,
      bodyStyle = props.bodyStyle,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);

  var _useState = (0, _react.useState)(props.isOpen),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      isOpen = _useState2[0],
      setIsOpen = _useState2[1];

  (0, _react.useEffect)(function () {
    if (props.isOpen !== isOpen) {
      setIsOpen(props.isOpen);
    }
  }, [props.isOpen]);

  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      loading = _useState4[0],
      setLoading = _useState4[1];

  var cls = [prefixCls, className, type ? "".concat(type) : null].filter(Boolean).join(' ').trim();

  function onClose() {
    setIsOpen(false);
  }

  function handleCancel(_x) {
    return _handleCancel.apply(this, arguments);
  }

  function _handleCancel() {
    _handleCancel = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(e) {
      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              setLoading(true);
              _context.prev = 1;
              _context.t0 = onCancel;

              if (!_context.t0) {
                _context.next = 6;
                break;
              }

              _context.next = 6;
              return onCancel(e);

            case 6:
              _context.next = 10;
              break;

            case 8:
              _context.prev = 8;
              _context.t1 = _context["catch"](1);

            case 10:
              setIsOpen(false);
              setLoading(false);

            case 12:
            case "end":
              return _context.stop();
          }
        }
      }, _callee, null, [[1, 8]]);
    }));
    return _handleCancel.apply(this, arguments);
  }

  function handleConfirm(_x2) {
    return _handleConfirm.apply(this, arguments);
  }

  function _handleConfirm() {
    _handleConfirm = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2(e) {
      return _regenerator["default"].wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              setLoading(true);
              _context2.prev = 1;
              _context2.t0 = onConfirm;

              if (!_context2.t0) {
                _context2.next = 6;
                break;
              }

              _context2.next = 6;
              return onConfirm(e);

            case 6:
              _context2.next = 10;
              break;

            case 8:
              _context2.prev = 8;
              _context2.t1 = _context2["catch"](1);

            case 10:
              setIsOpen(false);
              setLoading(false);

            case 12:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2, null, [[1, 8]]);
    }));
    return _handleConfirm.apply(this, arguments);
  }

  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactOverlay["default"], (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    usePortal: usePortal,
    isOpen: isOpen
  }, other), {}, {
    onClose: onClose,
    className: cls,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: "".concat(prefixCls, "-container"),
      children: /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
        className: ["".concat(prefixCls, "-inner"), title ? "".concat(prefixCls, "-shown-title") : null, icon ? "".concat(prefixCls, "-shown-icon") : null].filter(Boolean).join(' ').trim(),
        style: {
          maxWidth: maxWidth,
          minWidth: minWidth,
          width: width
        },
        children: [(title || icon) && /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
          className: "".concat(prefixCls, "-header"),
          children: [icon && /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactIcon["default"], {
            type: icon
          }), title && /*#__PURE__*/(0, _jsxRuntime.jsx)("h4", {
            children: title
          }), isCloseButtonShown && /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactButton["default"], {
            basic: true,
            onClick: function onClick(e) {
              return handleCancel(e);
            },
            icon: "close",
            type: "light"
          })]
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
          className: "".concat(prefixCls, "-body"),
          style: bodyStyle,
          children: children || content
        }), useButton && /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
          className: "".concat(prefixCls, "-footer"),
          children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_reactButton["default"], (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
            autoFocus: autoFocus,
            type: type,
            loading: loading,
            disabled: loading
          }, confirmButtonProps), {}, {
            onClick: function onClick(e) {
              return handleConfirm(e);
            },
            children: confirmText
          })), cancelText && /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactButton["default"], (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, cancelButtonProps), {}, {
            onClick: function onClick(e) {
              return handleCancel(e);
            },
            children: cancelText
          }))]
        })]
      })
    })
  }));
});

exports["default"] = _default;
module.exports = exports.default;
