// @uiw/react-loader/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "size", "loading", "tip", "vertical", "color", "bgColor", "children", "indicator", "fullscreen"];

var _default = function _default() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-loader' : _props$prefixCls,
      className = props.className,
      _props$size = props.size,
      size = _props$size === void 0 ? 'default' : _props$size,
      _props$loading = props.loading,
      loading = _props$loading === void 0 ? true : _props$loading,
      tip = props.tip,
      vertical = props.vertical,
      color = props.color,
      bgColor = props.bgColor,
      children = props.children,
      indicator = props.indicator,
      _props$fullscreen = props.fullscreen,
      fullscreen = _props$fullscreen === void 0 ? false : _props$fullscreen,
      otherProps = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var cls = [prefixCls, className, size ? "".concat(prefixCls, "-").concat(size) : null].filter(Boolean).join(' ').trim();
  var indicatorView = (0, _react.useMemo)(function () {
    return /*#__PURE__*/(0, _jsxRuntime.jsx)("svg", {
      viewBox: "25 25 50 50",
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)("circle", {
        cx: "50",
        cy: "50",
        r: "20",
        fill: "none",
        strokeWidth: "5",
        strokeMiterlimit: "10"
      })
    });
  }, []);
  var tipsView = (0, _react.useMemo)(function () {
    return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: ["".concat(prefixCls, "-tips"), fullscreen ? "".concat(prefixCls, "-fullscreen") : null].filter(Boolean).join(' ').trim(),
      style: {
        color: color,
        backgroundColor: bgColor
      },
      children: /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
        className: "".concat(prefixCls, "-tips-nested"),
        children: [indicator || indicatorView, tip && /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
          className: ["".concat(prefixCls, "-text"), vertical ? "".concat(prefixCls, "-vertical") : null].filter(Boolean).join(' ').trim(),
          children: tip
        })]
      })
    });
  }, [fullscreen, bgColor, prefixCls, vertical, tip]);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    className: cls
  }, otherProps), {}, {
    children: [(loading || fullscreen) && tipsView, children && /*#__PURE__*/_react["default"].cloneElement(children, Object.assign({}, children.props, {
      className: ["".concat(prefixCls, "-warp"), loading ? "".concat(prefixCls, "-blur") : null].filter(Boolean).join(' ').trim()
    }))]
  }));
};

exports["default"] = _default;
module.exports = exports.default;
