// @uiw/react-dropdown/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Dropdown;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _reactOverlayTrigger = _interopRequireDefault(require("@uiw/react-overlay-trigger"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "placement", "className", "menu", "children", "disabled"];

function Dropdown(props) {
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-dropdown' : _props$prefixCls,
      _props$placement = props.placement,
      placement = _props$placement === void 0 ? 'bottomLeft' : _props$placement,
      className = props.className,
      menu = props.menu,
      children = props.children,
      disabled = props.disabled,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var cls = (0, _react.useMemo)(function () {
    return [prefixCls, className].filter(Boolean).join(' ').trim();
  }, [prefixCls, className]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactOverlayTrigger["default"], (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    isOutside: true,
    autoAdjustOverflow: true,
    className: cls,
    disabled: disabled,
    placement: placement
  }, other), {}, {
    overlay: menu,
    children: /*#__PURE__*/_react["default"].cloneElement(children, Object.assign({}, children.props))
  }));
}

module.exports = exports.default;
