// @uiw/react-tag/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "style", "title", "children", "visible", "color", "disabled", "bordered", "closable", "light", "onClose"];

var _default = function _default() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-tag' : _props$prefixCls,
      className = props.className,
      style = props.style,
      _props$title = props.title,
      title = _props$title === void 0 ? '' : _props$title,
      children = props.children,
      _props$visible = props.visible,
      visible = _props$visible === void 0 ? true : _props$visible,
      _props$color = props.color,
      color = _props$color === void 0 ? '#6E6E6E' : _props$color,
      _props$disabled = props.disabled,
      disabled = _props$disabled === void 0 ? false : _props$disabled,
      _props$bordered = props.bordered,
      bordered = _props$bordered === void 0 ? true : _props$bordered,
      closable = props.closable,
      _props$light = props.light,
      light = _props$light === void 0 ? false : _props$light,
      onClose = props.onClose,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var cls = [prefixCls, className, light ? "".concat(prefixCls, "-light") : null, disabled ? 'disabled' : null].filter(Boolean).join(' ').trim();
  var styl = (0, _objectSpread2["default"])({}, style);

  if (!light) {
    styl.color = '#fff';
    styl.backgroundColor = color;
  } else {
    styl.color = color;
    styl.borderColor = color;

    if (bordered && light) {
      styl.boxShadow = "inset 0 0 0 1px ".concat(color);
    }
  }

  if (!visible) {
    return null;
  }

  var closeBtn = (0, _react.useMemo)(function () {
    return closable ? /*#__PURE__*/(0, _jsxRuntime.jsx)("svg", {
      onClick: onClose,
      className: "".concat(prefixCls, "-close"),
      width: "15",
      height: "15",
      viewBox: "0 0 16 16",
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)("path", {
        d: "M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 0 0-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 0 0-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 0 0 1.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71L9.41 8z"
      })
    }) : null;
  }, [closable]);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("span", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    className: cls,
    style: styl
  }, other), {}, {
    children: [title || children, closeBtn]
  }));
};

exports["default"] = _default;
module.exports = exports.default;
