// @uiw/react-menu/cjs/MenuItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _reactIcon = _interopRequireDefault(require("@uiw/react-icon"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "tagName", "children", "disabled", "multiline", "icon", "text", "active", "addonAfter", "isSubMenuItem"];
var disabledProps = {
  href: undefined,
  onClick: undefined,
  onMouseDown: undefined,
  onMouseEnter: undefined,
  onMouseLeave: undefined,
  tabIndex: -1
};

var MenuItem = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-menu-item' : _props$prefixCls,
      className = props.className,
      _props$tagName = props.tagName,
      TagName = _props$tagName === void 0 ? 'a' : _props$tagName,
      children = props.children,
      _props$disabled = props.disabled,
      disabled = _props$disabled === void 0 ? false : _props$disabled,
      _props$multiline = props.multiline,
      multiline = _props$multiline === void 0 ? false : _props$multiline,
      icon = props.icon,
      text = props.text,
      _props$active = props.active,
      active = _props$active === void 0 ? false : _props$active,
      addonAfter = props.addonAfter,
      isSubMenuItem = props.isSubMenuItem,
      htmlProps = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var anchorCls = [prefixCls, active ? 'active' : null, disabled ? 'w-disabled' : null, className].filter(Boolean).join(' ').trim();

  var tagComp = /*#__PURE__*/_react["default"].createElement(TagName, (0, _objectSpread2["default"])((0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, htmlProps), disabled ? disabledProps : {}), {}, {
    className: anchorCls,
    ref: ref
  }), /*#__PURE__*/(0, _jsxRuntime.jsxs)(_react.Fragment, {
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_reactIcon["default"], {
      className: "".concat(prefixCls, "-icon"),
      type: icon
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: [prefixCls && "".concat(prefixCls, "-text"), !multiline && "".concat(prefixCls, "-multiline")].filter(Boolean).join(' ').trim(),
      children: text
    }), addonAfter]
  }));

  if (isSubMenuItem) {
    return tagComp;
  }

  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("li", {
    children: [" ", tagComp, " "]
  });
});

MenuItem.displayName = 'uiw.MenuItem';
var _default = MenuItem;
exports["default"] = _default;
module.exports = exports.default;
