// @uiw/react-layout/cjs/Layout.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.LayoutContext = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "hasSider", "children"];

var LayoutContext = /*#__PURE__*/_react["default"].createContext({
  siderHook: {
    addSider: function addSider() {
      return null;
    },
    removeSider: function removeSider() {
      return null;
    }
  }
});

exports.LayoutContext = LayoutContext;

var Layout = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Layout, _React$Component);

  var _super = (0, _createSuper2["default"])(Layout);

  function Layout() {
    var _this;

    (0, _classCallCheck2["default"])(this, Layout);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "state", {
      siders: []
    });
    return _this;
  }

  (0, _createClass2["default"])(Layout, [{
    key: "getSiderHook",
    value: function getSiderHook() {
      var _this2 = this;

      return {
        addSider: function addSider(id) {
          _this2.setState(function (state) {
            return {
              siders: [].concat((0, _toConsumableArray2["default"])(state.siders), [id])
            };
          });
        },
        removeSider: function removeSider(id) {
          _this2.setState(function (state) {
            return {
              siders: state.siders.filter(function (currentId) {
                return currentId !== id;
              })
            };
          });
        }
      };
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          prefixCls = _this$props.prefixCls,
          className = _this$props.className,
          hasSider = _this$props.hasSider,
          children = _this$props.children,
          other = (0, _objectWithoutProperties2["default"])(_this$props, _excluded);
      return /*#__PURE__*/(0, _jsxRuntime.jsx)(LayoutContext.Provider, {
        value: {
          siderHook: this.getSiderHook()
        },
        children: /*#__PURE__*/(0, _jsxRuntime.jsx)("section", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
          className: [prefixCls, className, (typeof hasSider === 'boolean' ? hasSider : this.state.siders.length > 0) ? "".concat(prefixCls, "-has-sider") : null].filter(Boolean).join(' ').trim()
        }, other), {}, {
          children: children
        }))
      });
    }
  }]);
  return Layout;
}(_react["default"].Component);

exports["default"] = Layout;
(0, _defineProperty2["default"])(Layout, "Header", void 0);
(0, _defineProperty2["default"])(Layout, "Footer", void 0);
(0, _defineProperty2["default"])(Layout, "Sider", void 0);
(0, _defineProperty2["default"])(Layout, "Content", void 0);
(0, _defineProperty2["default"])(Layout, "defaultProps", {
  prefixCls: 'w-layout'
});
