// @uiw/react-layout/cjs/Sider.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _utils = require("@uiw/utils");

var _Layout = require("./Layout");

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "style", "children", "width", "collapsedWidth", "collapsed", "siderHook"];

function Sider() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-layout-sider' : _props$prefixCls,
      className = props.className,
      style = props.style,
      children = props.children,
      _props$width = props.width,
      width = _props$width === void 0 ? 200 : _props$width,
      _props$collapsedWidth = props.collapsedWidth,
      collapsedWidth = _props$collapsedWidth === void 0 ? 80 : _props$collapsedWidth,
      _props$collapsed = props.collapsed,
      collapsed = _props$collapsed === void 0 ? false : _props$collapsed,
      siderHook = props.siderHook,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);

  var _useState = (0, _react.useState)("w-layout-".concat((0, _utils.randomid)())),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 1),
      sliderId = _useState2[0];

  var _useState3 = (0, _react.useState)(collapsed ? collapsedWidth : width),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      rawWidth = _useState4[0],
      setRawWidth = _useState4[1];

  (0, _react.useEffect)(function () {
    if (siderHook && !!siderHook.addSider) {
      siderHook.addSider(sliderId);
    }

    return function () {
      if (siderHook && !!siderHook.removeSider) {
        siderHook.removeSider(sliderId);
      }
    };
  }, []);
  (0, _react.useMemo)(function () {
    setRawWidth(collapsed ? collapsedWidth : width);
  }, [collapsed]);
  var divStyle = (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, style), {}, {
    flex: "0 0 ".concat(rawWidth),
    maxWidth: rawWidth,
    minWidth: rawWidth,
    width: rawWidth
  });
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    className: [prefixCls, className].filter(Boolean).join(' ').trim(),
    style: divStyle
  }, other), {}, {
    children: children
  }));
}

var _default = function _default() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_Layout.LayoutContext.Consumer, {
    children: function children(context) {
      return /*#__PURE__*/(0, _jsxRuntime.jsx)(Sider, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, props), context));
    }
  });
};

exports["default"] = _default;
module.exports = exports.default;
