// @uiw/react-avatar/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _reactIcon = _interopRequireDefault(require("@uiw/react-icon"));

var _utils = require("@uiw/utils");

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "shape", "size", "className", "src", "alt", "icon", "onError"];

var _default = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-avatar' : _props$prefixCls,
      _props$shape = props.shape,
      shape = _props$shape === void 0 ? 'circle' : _props$shape,
      _props$size = props.size,
      size = _props$size === void 0 ? 'default' : _props$size,
      className = props.className,
      src = props.src,
      alt = props.alt,
      icon = props.icon,
      _props$onError = props.onError,
      _onError = _props$onError === void 0 ? _utils.noop : _props$onError,
      resetProps = (0, _objectWithoutProperties2["default"])(props, _excluded);

  var children = props.children;

  var _useState = (0, _react.useState)(true),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      isImgExist = _useState2[0],
      setIsImgExist = _useState2[1];

  var cls = [prefixCls, className, size ? "".concat(prefixCls, "-").concat(size) : null, shape ? "".concat(prefixCls, "-").concat(shape) : null, src ? "".concat(prefixCls, "-image") : null].filter(Boolean).join(' ').trim();
  (0, _react.useEffect)(function () {
    setIsImgExist(true);
  }, [props.src]);

  if (isImgExist && src) {
    children = /*#__PURE__*/(0, _jsxRuntime.jsx)("img", {
      src: src,
      alt: alt,
      onError: function onError(evn) {
        var errorFlag = _onError ? _onError(evn) : undefined;

        if (errorFlag !== false) {
          setIsImgExist(false);
        }
      }
    });
  } else if (icon && typeof icon === 'string') {
    children = /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactIcon["default"], {
      type: icon
    });
  } else if (icon && /*#__PURE__*/_react["default"].isValidElement(icon)) {
    children = icon;
  }

  return /*#__PURE__*/(0, _jsxRuntime.jsx)("span", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, resetProps), {}, {
    className: cls,
    ref: ref,
    children: children
  }));
});

exports["default"] = _default;
module.exports = exports.default;
