// @uiw/react-notify/cjs/Container.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _reactAlert = _interopRequireDefault(require("@uiw/react-alert"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["description", "isOpen"];
var notifys = {};
var timer = {};

var Container = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Container, _React$Component);

  var _super = (0, _createSuper2["default"])(Container);

  function Container() {
    var _this;

    (0, _classCallCheck2["default"])(this, Container);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "state", {
      notifys: {}
    });
    return _this;
  }

  (0, _createClass2["default"])(Container, [{
    key: "create",
    value: function create(props) {
      var _this2 = this;

      var placement = props.placement,
          key = props.key;

      if (!notifys[placement]) {
        notifys[placement] = {};
      }

      props.isOpen = false;
      notifys[placement][key] = props;

      if (props.duration) {
        timer[key] = setTimeout(function () {
          _this2.closed(key, placement);
        }, props.duration);
      }

      this.setState({
        notifys: notifys,
        placement: placement
      }, function () {
        notifys[placement][key].isOpen = true;

        _this2.setState({
          notifys: notifys
        });
      });
    }
  }, {
    key: "closed",
    value: function closed(key, placement) {
      if (!key || !placement || !notifys[placement][key]) {
        return;
      }

      notifys[placement][key].isOpen = false;
      var props = notifys[placement][key];
      this.setState({
        notifys: notifys
      }, function () {
        clearTimeout(timer[key]);
        delete timer[key];
        delete notifys[placement][key];

        if (props && props.willUnmount) {
          props.willUnmount(props, notifys);
        }
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this3 = this;

      var prefixCls = this.props.prefixCls;
      var placement = this.state.placement;
      return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react["default"].Fragment, {
        children: placement && Object.keys(this.state.notifys[placement]).map(function (key) {
          var _this3$state$notifys$ = _this3.state.notifys[placement][key],
              description = _this3$state$notifys$.description,
              isOpen = _this3$state$notifys$.isOpen,
              alertProps = (0, _objectWithoutProperties2["default"])(_this3$state$notifys$, _excluded);

          if (alertProps.type === 'open') {
            delete alertProps.type;
          }

          return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactAlert["default"], (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
            className: prefixCls,
            useButton: false,
            width: 320
          }, alertProps), {}, {
            usePortal: false,
            hasBackdrop: false,
            isOpen: isOpen,
            content: description
          }), key);
        })
      });
    }
  }]);
  return Container;
}(_react["default"].Component);

exports["default"] = Container;
(0, _defineProperty2["default"])(Container, "defaultProps", {
  prefixCls: 'w-notify',
  placement: 'topRight'
});
module.exports = exports.default;
