// @uiw/react-rate/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Rate;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _utils = require("@uiw/utils");

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "count", "value", "className", "allowHalf", "character", "readOnly", "disabled", "onChange", "onHoverChange", "color"];

function Rate() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-rate' : _props$prefixCls,
      _props$count = props.count,
      count = _props$count === void 0 ? 5 : _props$count,
      _props$value = props.value,
      defValue = _props$value === void 0 ? 0 : _props$value,
      className = props.className,
      allowHalf = props.allowHalf,
      _props$character = props.character,
      character = _props$character === void 0 ? '★' : _props$character,
      _props$readOnly = props.readOnly,
      readOnly = _props$readOnly === void 0 ? false : _props$readOnly,
      disabled = props.disabled,
      _props$onChange = props.onChange,
      onChange = _props$onChange === void 0 ? _utils.noop : _props$onChange,
      _props$onHoverChange = props.onHoverChange,
      onHoverChange = _props$onHoverChange === void 0 ? _utils.noop : _props$onHoverChange,
      color = props.color,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);

  var _useState = (0, _react.useState)(defValue),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      value = _useState2[0],
      setValue = _useState2[1];

  var _useState3 = (0, _react.useState)(-1),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      hoverCount = _useState4[0],
      setHoverCount = _useState4[1];

  var cls = [prefixCls, className, disabled ? 'disabled' : null].filter(Boolean).join(' ').trim();

  var _useState5 = (0, _react.useState)(),
      _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
      prevValue = _useState6[0],
      setPrevValue = _useState6[1];

  if (defValue !== prevValue) {
    setPrevValue(defValue);
  }

  (0, _react.useMemo)(function () {
    if (value !== prevValue) {
      setValue(defValue);
    }
  }, [prevValue]);

  function _onMouseLeave() {
    setHoverCount(-1);
  }

  function getValue(e, key) {
    e.persist();
    var currentValue = key;
    var isLeft = e.clientX - e.currentTarget.getBoundingClientRect().left <= e.currentTarget.getBoundingClientRect().width / 2;

    if (allowHalf) {
      e.persist();
      currentValue = isLeft ? key + 0.5 : key + 1;
    } else {
      currentValue = key + 1;
    }

    return currentValue;
  }

  function onMouseMove(e, key) {
    var currentValue = getValue(e, key);

    if (hoverCount !== currentValue) {
      setHoverCount(currentValue);
      onHoverChange(currentValue);
    }
  }

  function onClick(e, key) {
    if (readOnly) return;
    var currentValue = getValue(e, key);
    setValue(currentValue);
    onChange(currentValue);
  }

  return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, other), {}, {
    className: cls,
    onMouseLeave: function onMouseLeave() {
      return _onMouseLeave();
    },
    children: (0, _toConsumableArray2["default"])(Array(count)).map(function (_, idx) {
      var halfon = value <= idx + 0.5 && Math.ceil(value) - 1 === idx && hoverCount === -1 || hoverCount === idx + 0.5;
      var activeCls = ["".concat(prefixCls, "-hight"), idx + 1 <= value && hoverCount === -1 ? 'star-on' : null, idx + 1 <= hoverCount ? 'hover-on' : null, halfon ? 'half-on' : null].filter(Boolean).join(' ').trim();
      var props = {};

      if (!readOnly) {
        props.onClick = function (e) {
          return onClick(e, idx);
        };

        props.onMouseMove = function (e) {
          return onMouseMove(e, idx);
        };
      }

      return /*#__PURE__*/(0, _jsxRuntime.jsxs)("span", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, props), {}, {
        children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
          style: {
            color: color
          },
          className: activeCls,
          children: character
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
          className: "".concat(prefixCls, "-bg"),
          children: character
        })]
      }), idx);
    })
  }));
}

module.exports = exports.default;
