// @uiw/react-tooltip/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _typeof2 = _interopRequireDefault(require("@babel/runtime/helpers/typeof"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _reactOverlayTrigger = _interopRequireDefault(require("@uiw/react-overlay-trigger"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "placement", "isOpen", "trigger", "delay", "usePortal", "visibleArrow", "onVisibleChange"];

var _default = function _default() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-tooltip' : _props$prefixCls,
      className = props.className,
      _props$placement = props.placement,
      placement = _props$placement === void 0 ? 'top' : _props$placement,
      _props$isOpen = props.isOpen,
      isOpen = _props$isOpen === void 0 ? false : _props$isOpen,
      trigger = props.trigger,
      delay = props.delay,
      _props$usePortal = props.usePortal,
      usePortal = _props$usePortal === void 0 ? true : _props$usePortal,
      _props$visibleArrow = props.visibleArrow,
      visibleArrow = _props$visibleArrow === void 0 ? true : _props$visibleArrow,
      onVisibleChange = props.onVisibleChange,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var cls = [prefixCls, className, !visibleArrow ? 'no-arrow' : null].filter(Boolean).join(' ').trim();
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactOverlayTrigger["default"], (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    usePortal: usePortal,
    isOpen: isOpen,
    trigger: trigger,
    delay: delay,
    onVisibleChange: onVisibleChange,
    placement: placement
  }, other), {}, {
    overlay: /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
      className: cls,
      children: [visibleArrow && /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        className: "".concat(prefixCls, "-arrow")
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        className: "".concat(prefixCls, "-inner"),
        children: props.content
      })]
    }),
    children: (0, _typeof2["default"])(props.children) === 'object' ? props.children : /*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
      children: props.children
    })
  }));
};

exports["default"] = _default;
module.exports = exports.default;
